ui/src/utils/QualityControl.js

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Responsible for quality control  
+ */
+export class QualityControl {
+    
+   /**
+    * Creates an quality control instance
+    */
+    constructor(context) {
+        this.context = context;
+    }
+    
+   /**
+    * Checks if sound is of good quality
+    */
+    isQualitySound() {
+        if(this.lengthCheck()) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+   /**
+    * Checks if audio is too short
+    */
+    lengthCheck() {
+        if (this.context.currentTime < 2.0) {
+            alert('Audio recording failed: recording was too short. Try again');
```

Comment:
We should create a simple API that indicates success or failure with error attached. So we can add additional checks which conform to the same interface, making it easier to collect errors.

Something like

```javascript
do.some.check();
if (success) {
  return null;
} else {
  return 'Needs more cowbell',
}
```

So then in `isQualityAudio` we can make each of these calls and decide if it's a success, or collect all the errors to present to the user, in our toast component.

```javascript
isQualityAudio() {
  // above we set up so that lengthCheck returns null unless there is an error
  // set up a reporting object to describe errors to caller
  const qualityResult = {
    success: true,
    errorMessage: '',
  }

  // probably a more clean way to do this, but something to this effect
  const lengthResult = this.lengthCheck();
  if (lengthResult) {
    qualityResult.success = false;
    qualityResult.errorMessage += lengthResult;
  }
  const otherResult = this.otherCheck();
   if (otherResult) {
    qualityResult.success = false;
    qualityResult.errorMessage += otherResult;
  }

  return qualityResult;
}

Reply:
This error will not go into the toast component. Eldrick is working on showing the audio QC errors on top of the button. We thought it would be easier for the user to only display one error even if there are multiple instead of throwing them all at once so this will most likely become a bunch of else ifs depending on the amount of errors

Reply:
As I understand it, the toast is for javascript errors. This quality check message will do two things 1) show the message above the record button and 2) preventing the user from submitting.

Reply:
Also, we'll be showing one message at a time right? And since there are only two cases at the moment, it seems like overengineering to create an API for quality errors.

Also if you += to qualityResult.errorMessage without returning after each if statement you'll get a concatenated errorMessage.